### PR TITLE
replace redhawk with nativeaot

### PR DESCRIPTION
**Pull request by kasperk81**
only mentions of redhawk left are in nativeaot directory

### CODE REVIEW

File: src/coreclr/tools/Common/Internal/Runtime/MethodTable.Constants.cs [hunk 1]
```diff
@@ -200,7 +200,7 @@ internal enum EETypeField
         ETF_DynamicThreadStaticOffset,
     }
 
-    // Subset of the managed TypeFlags enum understood by Redhawk.
+    // Subset of the managed TypeFlags enum understood by NativeAOT.
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/Common/Internal/Runtime/MethodTable.Constants.cs:203]:
```suggestion
    // Subset of the managed TypeFlags enum understood by the runtime.
```

File: src/tests/GC/Scenarios/BaseFinal/basefinal.cs [hunk 2]
```diff
@@ -90,7 +90,7 @@ public CreateObj()
         {
 
             mv_Obj = new BaseFinal();
-            //Printing GetTotalMemory is commented out so that the test can build on RedHawk
+            //Printing GetTotalMemory is commented out so that the test can build on NativeAOT
```
Comment (MichalStrehovsky) [inline src/tests/GC/Scenarios/BaseFinal/basefinal.cs:93]: This is probably a very old comment referring to NativeAOT without a real corelib. We could delete the comment and uncomment the line.

File: src/coreclr/inc/cordebuginfo.h [hunk 3]
```diff
@@ -192,11 +192,11 @@ class ICorDebugInfo
 #elif TARGET_AMD64
         REGNUM_SP = REGNUM_RSP,
 #elif TARGET_ARM
-#ifdef REDHAWK
+#ifdef FEATURE_NATIVEAOT
```
Comment (MichalStrehovsky) [inline src/coreclr/inc/cordebuginfo.h:195]: This file doesn't compile into NativeAOT. This is some very old define. I would delete it, I don't even know if it's true.

Refreshing the name from REDHAWK to NativeAOT would make it look like the information is fresh.

File: src/coreclr/gc/gc.cpp [hunk 4]
```diff
@@ -34777,7 +34777,7 @@ BOOL gc_heap::commit_mark_array_bgc_init()
             {
                 // For ro segments they could always be only partially in range so we'd
                 // be calling this at the beginning of every BGC. We are not making this
-                // more efficient right now - ro segments are currently only used by redhawk.
+                // more efficient right now - ro segments are currently only used by NativeAOT.
```
Comment (MichalStrehovsky) [inline src/coreclr/gc/gc.cpp:34780]: @dotnet/gc this comment is obviously not true in 2022. How should this comment be written?

File: src/coreclr/gc/env/gcenv.base.h [hunk 5]
```diff
@@ -485,7 +485,7 @@ class Object;
 class ArrayBase;
 
 // Various types used to refer to object references or handles. This will get more complex if we decide
-// Redhawk wants to wrap object references in the debug build.
+// NativeAOT wants to wrap object references in the debug build.
```
Comment (MichalStrehovsky) [inline src/coreclr/gc/env/gcenv.base.h:488]: This comment doesn't look useful. I would just delete.

File: src/coreclr/nativeaot/Runtime/rheventtrace.cpp [hunk 6]
```diff
@@ -18,7 +18,6 @@
 #include "shash.h"
 #include "eventtracepriv.h"
 #include "shash.inl"
-#include "palredhawk.h"
```
Comment (kasperk81) [inline src/coreclr/nativeaot/Runtime/rheventtrace.cpp:21]: @MichalStrehovsky, if `palredhawk.h` doesn't exist in the repo or generated by cmake or a script, how does `rheventtrace.cpp` compile on windows?

File: src/coreclr/gc/env/gcenv.base.h (hunk 5 above)
Comment (kasperk81) [reply src/coreclr/gc/env/gcenv.base.h:488]:
```suggestion
```

File: src/coreclr/gc/env/gcenv.base.h [hunk 7]
```diff
@@ -485,7 +485,7 @@ class Object;
 class ArrayBase;
 
 // Various types used to refer to object references or handles. This will get more complex if we decide
```
Comment (kasperk81) [inline src/coreclr/gc/env/gcenv.base.h:487]:
```suggestion
```

File: src/coreclr/inc/cordebuginfo.h [hunk 8]
```diff
@@ -192,11 +192,11 @@ class ICorDebugInfo
...
         REGNUM_FP = REGNUM_R7,
 #else
```
Comment (kasperk81) [inline src/coreclr/inc/cordebuginfo.h:197]:
```suggestion
```

File: src/coreclr/inc/cordebuginfo.h [hunk 9]
```diff
@@ -192,11 +192,11 @@ class ICorDebugInfo
...
         REGNUM_FP = REGNUM_R11,
-#endif //REDHAWK
+#endif //FEATURE_NATIVEAOT
```
Comment (kasperk81) [inline src/coreclr/inc/cordebuginfo.h:199]:
```suggestion
```

File: src/coreclr/nativeaot/Runtime/rheventtrace.cpp (hunk 6 above)
Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/Runtime/rheventtrace.cpp:21]: https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/Runtime/PalRedhawk.h

File: src/tests/GC/Scenarios/BaseFinal/basefinal.cs [hunk 10]
```diff
@@ -90,7 +90,7 @@ public CreateObj()
...
             //Console.WriteLine("before test started, the heapsize is {0}", GC.GetTotalMemory(false));
```
Comment (kasperk81) [inline src/tests/GC/Scenarios/BaseFinal/basefinal.cs:94]:
```suggestion
            Console.WriteLine("before starting the test, heap size is {0}", GC.GetTotalMemory(false));
```

File: src/coreclr/nativeaot/Runtime/rheventtrace.cpp (hunk 6 above)
Comment (kasperk81) [reply src/coreclr/nativeaot/Runtime/rheventtrace.cpp:21]: i see. different casing. i should've used `find -iname`  :)

it built without this include

**Review by agocke — APPROVED** [review]
LGTM. If we want to make further changes, I think we can do it in a separate PR.

**Pull request merged**